### PR TITLE
refactor(yaml): cleanup dead code in `Mark` class

### PR DESCRIPTION
**Pull request by iuioiua**

### CODE REVIEW

File: yaml/_mark.ts [hunk 1]
```diff
@@ -20,59 +24,42 @@ export class Mark {
...
-    if (!compact) {
-      snippet = this.getSnippet();
-
-      if (snippet) {
-        where += `:\n${snippet}`;
-      }
-    }
-
-    return where;
+  toString(): string {
```
Comment (iuioiua) [inline yaml/_mark.ts:61]: The `compact` argument is never used.

File: yaml/_mark.ts [hunk 2]
```diff
@@ -20,59 +24,42 @@ export class Mark {
     this.column = column;
   }
 
-  getSnippet(indent = 4, maxLength = 75): string | null {
-    if (!this.buffer) return null;
```
Comment (kt3k) [inline yaml/_mark.ts:24]: Is it safe to remove this check?

Comment (iuioiua) [reply yaml/_mark.ts:24]: Realistically, this is safe to remove, unless the user disables or misuses TypeScript. Either way, I've reverted this for now.

File: yaml/_mark.ts [hunk 3]
```diff
@@ -20,59 +24,43 @@ export class Mark {
...
-  toString(compact?: boolean): string {
-    let snippet;
-    let where = "";
-
-    where += `at line ${this.line + 1}, column ${this.column + 1}`;
-
-    if (!compact) {
-      snippet = this.getSnippet();
-
-      if (snippet) {
```
Comment (kt3k) [inline yaml/_mark.ts:71]: I think now we need to keep this condition

**Comment by timreichen** [comment]
I kinda think we could just make `Mark` an interface inside `_error.ts` and combine `getSnippet()` and `toString()` into one `stringifyMark()` function. The mark class is not use elsewhere and the instance is only used once inside the error constructor...

**Comment by iuioiua** [comment]
> I kinda think we could just make `Mark` an interface inside `_error.ts` and combine `getSnippet()` and `toString()` into one `stringifyMark()` function. The mark class is not use elsewhere and the instance is only used once inside the error constructor...

Yeah, let's do that in a following PR.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**